punchcut/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+/*!
+Glyph loading.
+*/
+
+#![forbid(unsafe_code)]
+// TODO: this is temporary-- remove when hinting is added.
+#![allow(dead_code, unused_imports, unused_variables)]
+
+/// Re-export of peniko crate.
+pub use peniko;
+
+mod error;
+mod outline;
+mod scaler;
+
+pub mod source;
+
+/// Representations of fonts and font collections.
+pub mod font {
+    pub use read_fonts::{types::Tag, CollectionRef, FileRef, FontRef, TableProvider};
+}
+
+use font::Tag;
+use source::*;
+
+use core::str::FromStr;
+
+pub use error::*;
+pub use outline::*;
+pub use scaler::*;
+
+/// Modes for hinting.
+///
+/// Only the `glyf` source supports all hinting modes.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+#[allow(dead_code)]
+pub enum Hinting {
+    /// "Full" hinting mode. May generate rough outlines and poor horizontal
+    /// spacing.
+    Legacy,
+    /// Subpixel mode for grayscale rendering.
+    GrayscaleSubpixel,
+    /// Cleartype subpixel mode.
+    Subpixel,
+    /// Same as subpixel, but always prevents adjustment in the horizontal
+    /// direction. This is the default mode.
+    #[default]
+    Modern,
+}
+
+/// Type for a normalized variation coordinate.
+pub type NormalizedCoord = read_fonts::types::F2Dot14;
+
+/// Type for a glyph identifier.
+pub type GlyphId = read_fonts::types::GlyphId;
+
+/// Setting for specifying a variation by tag and value.
+#[derive(Copy, Clone, Debug)]
+pub struct Variation {
+    /// Tag for the variation.
+    pub tag: Tag,
+    /// Value for the variation.
+    pub value: f32,
+}
+
+impl From<(Tag, f32)> for Variation {
+    fn from(s: (Tag, f32)) -> Self {
+        Self {
+            tag: s.0,
+            value: s.1,
+        }
+    }
+}
+
+impl From<(&str, f32)> for Variation {
+    fn from(s: (&str, f32)) -> Self {
+        Self {
+            tag: Tag::from_str(s.0).unwrap_or_default(),
+            value: s.1,
+        }
+    }
+}
+
+impl From<([u8; 4], f32)> for Variation {
+    fn from(s: ([u8; 4], f32)) -> Self {
+        Self {
+            tag: Tag::new_checked(&s.0[..]).unwrap_or_default(),
+            value: s.1,
+        }
+    }
+}
+
+/// Context for loading glyphs from any available source.
+pub struct Context {
+    /// Inner context for loading TrueType outlines.
+    glyf: glyf::Context,
+    /// Internal storage for TrueType outlines.
+    glyf_outline: glyf::Outline,
+    /// Storage for normalized variation coordinates.
+    coords: Vec<NormalizedCoord>,
+    /// Storage for variation settings.
+    variations: Vec<Variation>,
+}
+
+impl Default for Context {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Context {
+    /// Creates a new glyph loading context.
+    pub fn new() -> Self {
+        Self {
+            glyf: glyf::Context::new(),
+            glyf_outline: glyf::Outline::new(),
+            coords: vec![],
+            variations: vec![],
+        }
```

Comment:
this looks like it should be able to derive default?

Reply:
I believe so. I'll add it.